contracts/Manager.sol

```diff
@@ -617,4 +630,85 @@ contract Manager is UUPSOwnableUpgradeable, UsingRegistryUpgradeable {
             }
         }
     }
+
+    /**
+     * @notice Sets vote contract address.
+     * @param _voteContract Vote contract address.
+     */
+    function setVoteContract(address _voteContract) public onlyOwner {
+        require(_voteContract != address(0), "Null address");
+        voteContract = _voteContract;
+        emit VoteContractSet(_voteContract);
+    }
+
+    /**
+     * @notice Votes on a proposal in the referendum stage.
+     * @param proposalId The ID of the proposal to vote on.
+     * @param index The index of the proposal ID in `dequeued`.
+     * @param yesVotes The yes votes weight.
+     * @param noVotes The no votes weight.
+     * @param abstainVotes The abstain votes weight.
+     */
+    function voteProposal(
+        uint256 proposalId,
+        uint256 index,
+        uint256 yesVotes,
+        uint256 noVotes,
+        uint256 abstainVotes
+    ) public {
+        IVote vote = IVote(voteContract);
+
+        (
+            uint256 stakedCeloBalance,
+            uint256 totalYesVotes,
+            uint256 totalNoVotes,
+            uint256 totalAbstainVotes
+        ) = vote.voteProposal(msg.sender, proposalId, yesVotes, noVotes, abstainVotes);
+
+        stakedCelo.lockBalance(msg.sender, stakedCeloBalance);
+        account.voteProposal(proposalId, index, totalYesVotes, totalNoVotes, totalAbstainVotes);
+    }
+
+    /**
+     * @notice Revokes votes on already voted proposal.
+     * @param proposalId The ID of the proposal to vote on.
+     * @param index The index of the proposal ID in `dequeued`.
+     */
+    function revokeVotes(uint256 proposalId, uint256 index) external {
+        IVote vote = IVote(voteContract);
+
+        (, uint256 totalYesVotes, uint256 totalNoVotes, uint256 totalAbstainVotes) = vote
+            .revokeVotes(msg.sender, proposalId);
+
+        account.voteProposal(proposalId, index, totalYesVotes, totalNoVotes, totalAbstainVotes);
+    }
+
+    /**
+     * @notice Unlock balance of stCelo.
+     * @param beneficiary The account to be unlocked.
+     */
+    function updateHistoryAndReturnLockedStCeloInVoting(address beneficiary)
+        external
+        returns (uint256)
+    {
+        IVote vote = IVote(voteContract);
+        return vote.updateHistoryAndReturnLockedStCeloInVoting(beneficiary);
+    }
+
+    /**
+     * @notice Unlock balance of stCelo.
+     * @param accountAddress The account to be unlocked.
+     */
+    function unlockBalance(address accountAddress) public {
+        stakedCelo.unlockBalance(accountAddress);
+    }
+
+    /**
+     * @notice Overrides locked stBalance.
+     * @param accountAddress The account to be overriden.
+     * @param newLockBalance The new locked balance.
+     */
+    function overrideLockBalance(address accountAddress, uint256 newLockBalance) public onlyOwner {
```

Comment:
In what case would we want to manually override a users locked balance?

Reply:
This was suggested to implement by @martinvol

Reply:
The reasoning was that in case something changes in the lockedGold contract, and likely this contract wont be aware of that, so this is an emergency way of unlocking funds. As the owner can deploy any code, this actually doesn't hurt in normal situations.